internal/http/services/owncloud/ocs/handlers/apps/sharing/shares/public.go

```diff
@@ -457,23 +464,30 @@ func (h *Handler) updatePublicShare(w http.ResponseWriter, r *http.Request, shar
 	}
 
 	// Password
-	newPassword, ok := r.Form["password"]
+	_, ok = r.Form["password"]
+	newPassword := strings.TrimSpace(r.FormValue("password"))
 	// enforcePassword
 	if h.enforcePassword(permKey) {
-		if (!ok && !share.PasswordProtected) || (ok && len(strings.TrimSpace(newPassword[0])) == 0) {
+		if !ok && !share.PasswordProtected || ok && len(newPassword) == 0 {
 			response.WriteOCSError(w, r, response.MetaBadRequest.StatusCode, "missing required password", err)
 			return
 		}
 	}
 
+	// skip validation if the clear password scenario
+	if err := h.passwordValidator.Validate(newPassword); ok && len(newPassword) > 0 && err != nil {
```

Comment:
`len(newPassword) > 0` already implies that `ok == true` so the check for `ok` can be omited. Also same comment as above: Check `len(password) > 0` first because there is no need to validate the password if there is none.